module/VuFind/src/VuFind/Db/Service/ResourceService.php

```diff
@@ -109,18 +110,42 @@ public function createEntity(): ResourceEntityInterface
     }
 
     /**
-     * Get a set of records that do not have metadata stored in the resource
-     * table.
+     * Get a set of records that are missing metadata in the resource table. If maxAge is specified, this includes also
+     * records that need to be updated.
+     *
+     * @param ?int  $lastId  ID of last checked record, or null to start from beginning
+     * @param int   $limit   Limit for results
+     * @param ?int  $minAge  Minimum age (in days) for metadata before it needs to be updated, or null to search for
+     * records that are missing metadata
+     * @param array $sources Record source filter
```

Comment:
You could make this type more specific:


```suggestion
     * @param string[] $sources Record source filter
```